CHANGELOG.md

```diff
@@ -16,6 +16,25 @@ Versioning](https://semver.org/spec/v2.0.0.html).
   is less than one second.
 - `init_expiration_time` and `update_jwt_expires_in` take `u32` instead of `i64`
   for the expiration time argument.
+- `Node` struct now has `setting` and `setting_draft` of type `NodeSetting`, and
+  `name` and `name_draft`. Upon initial insertion of `Node`, `name` must be
+  provided, as it is used as the key of `Node` in the database. `name_draft` and
+  `setting_draft` are introduced to support 2-step node-setting process, which
+  is save & apply. `name_draft` and `setting_draft` fields mean that the data
+  are only saved to the database. Once those are applied, the draft values are
+  moved to `name`, and `setting`.
+  - Renamed `updateNode` GraphQL API to `updateNodeDraft`, and modified
+    parameter types. `old` to `NodeInput`, and `new` to `NodeDraftInput`,
+    respectively with and without `setting` field.
+  - Introduced `applyNode` GraphQL API, that applies draft values to modules and
+    updates values in database. This API handles partial success of setting
+    application, which may happen when a node carries multiple modules.
+    `ApplyResult::success_modules` lists the succeeded modules' names.
+  - `graphql::event::convert_sensors` uses `Node`'s `setting` value, to retrieve
+    the hostnames of the sensors. This function is called by GraphQL APIs of
+    `EventQuery` and `EventGroupQuery`.
+  - `node_status_list` API uses `hostname` from `Node`'s `setting` field.
+  - `graphql::node::crud::get_node_settings` uses `Node`'s `setting` value.
```

Comment:
`get_node_settings` is newly added not changed.

Reply:
While `Node` currently uses the singular term `setting`, the function for accessing it has been named `get_node_settings` ending with the plural term `settings`, indicating a need for consistent terminology. I recommend using `settings` instead of `setting` since it reflects the presence of multiple configurations, not merely one. Consequently, it would be more appropriate to replace `setting` with `settings` within `Node`, and rename `NodeSetting` to `NodeSettings` for clarity and accuracy.

Reply:
With the recent big changes to node-related things, do we still need this `getNodeSettings` function?